src/stirling/source_connectors/perf_profiler/testing/java/ProfilerTest.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2018- The Pixie Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+class ProfilerTest {
+  // Our two leaf functions do not share their code, as we had originally intended. Sharing
+  // would work like this:
+  // leaf2x() { return leaf(5);  }  // Crafted such that leaf(5) does 2x the work of leaf(10).
+  // leaf1x() { return leaf(10); }
+  // i.e. with a shared parameterized leaf function, leaf1x() and leaf2x() are just pass throughs.
+  // Unfortunately, different JVMs inline functions in different ways. Some of them would inline
+  // leaf1x and leaf2x and as a result, we were unable to find the expected symbols.
+
+  public static long leaf1x() {
```

Comment:
Is there a reason we can't just sleep for 5 seconds in one and 10 seconds in the other?

Reply:
Does sleep just de-schedule the process? If so, then we probably wouldn't actually see our process doing much at all. Also, this gives us control over the symbols (i.e. versus whatever specific impl. of sleep comes with the multitude of JVMs that we may test with).

Reply:
I suppose sleep wouldn't quite work but something like:
```
long startTime = System.currentTimeMillis();

while ((System.curentTimeMillis() - startTime) < 5*1000) {
}
```
seems a lot simpler to me. As it is there's `m`, `n`, `s`, `i`, `j`, and `k` and its quite hard to understand what's going on.

In terms of the stack traces, couldn't we just ignore anything below `leaf1x` or `leaf2x` in the stack.

Reply:
We removed some of the seemingly useless code and added a better comment to explain what remains.

In more detail, we ran a few tests to evaluate different versions of `ProfilerTest.java`.
1. we tested using the call to `System.currentTimeMillis()`.
2. we tested both with and without the "dead" and "useless" code in the loop.

For experiment (1) we found that use of the system call caused stack traces to get "broken": when the stack trace is sampled inside of the clock system call, it does not get connected to its caller, i.e. the system call to clock is not connected to its calling function leaf1x or leaf2x. The test expectations were met, but with fewer samples and more variance. 

For experiment (2), we removed the "prime number mod" in the loop. We found that some JVMs were able to statically analyze the very simple code and create a highly optimized representation of the function. For example, without the "dead code" in the loop, for our "azul-zulu-alipine" image, we found that leaf1x and leaf2x did the same amount of work and that they had a 10x speedup versus their "harder to optimize version" that we are merging in this PR. We want the "harder to optimize" version because we want leaf2x and leaf1x to do the work so that our test expectation is met.